lib/TransferJobs/SubDirectoryTransfer.cs

```diff
@@ -350,11 +350,19 @@ private void CreateDestinationDirectory(CancellationToken cancellationToken)
                 }
                 catch (StorageException storageException)
                 {
-                    throw new TransferException(TransferErrorCode.FailToVadlidateDestination,
-                        string.Format(CultureInfo.CurrentCulture,
-                            Resources.FailedToValidateDestinationException,
-                            storageException.ToErrorDetail()),
-                        storageException);
+                    if ((null != storageException.InnerException)
```

Comment:
Would you like to look at other place to see it the OperationCanceledException also wrappered to StorageException by XSCL.

Reply:
The change here is for our own exception handling.
In HierarchyDirectoryTransfer, for it lists source directory and its subdirectories with multiple threads, when there's any error happens in any one list threading, DMLib will try to stop the whole HierarchyDirectoryTransfer with cancelling, throw out the origin exception which causes the error. Thus DMLib would need to ignore OperationCancelled exception (which is caused by ourselves), and throw out other StorageException (which may represent the actual error root cause).

Reply:
For StorageException needs to thrown out in other place, DMLib's current behavior is to directly throw it out without handling. So we may not need to change it.